src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8578,33 +8577,67 @@ static bool isCandidateUnique(ref MethodSymbol? method, MethodSymbol candidate)
         {
             Debug.Assert(returnTypeOpt.Type?.IsVoidType() != true); // expecting !returnTypeOpt.HasType rather than System.Void
 
-            if (returnRefKind == RefKind.None &&
-                (parameterRefKinds.IsDefault || parameterRefKinds.All(refKind => refKind == RefKind.None)))
+            bool returnsVoid = !returnTypeOpt.HasType;
+            var typeArguments = returnsVoid ? parameterTypes : parameterTypes.Add(returnTypeOpt);
+
+            if (returnsVoid && returnRefKind != RefKind.None)
+            {
+                // Invalid return type.
+                return null;
```

Comment:
Do we want to consider diagnostics for the cases we're returning `null`? It would be better to have specific error messages for those cases, I think.

Reply:
A specific error shouldn't be necessary. There are two cases where we return `null`: the case here of a `ref void` return type which will have been reported as a declaration error; and the case below where a parameter or return type cannot be used as a type argument. The second case should be resolved in [https://github.com/dotnet/roslyn/issues/55217](https://github.com/dotnet/roslyn/issues/55217).